cleanco/classify.py

```diff
@@ -3,59 +3,69 @@
 
 Examples of use:
 
->> # check name for its possible business type(s)
->> classification_sources = typesources()
->> matches("MyCompany Ltd", classification_sources)
-['Limited']
->>
-
->> # check name for its possible jurisdictions, usually countries
->> classification_sources = countrysources()
->> matches("MyCompany Ltd", classification_sources)
-['New Zealand', 'United Kingdom', 'United States of America']
->>
+# check name for its possible business type(s)
+>> matches("MyCompany Ltd", "local_name")
+['Private Limited by Guarantee', 'Private Limited Company', 'Limited Liability Company', 'New Generation Co-operatives']
+
+# check name for its possible countries
+>> matches("MyCompany Ltd", "country")
+['United States of America', 'United Kingdom', 'Canada']
 
 """
 
-from .termdata import terms_by_country, terms_by_type
+from iso20275 import Elf, ElfEntry
 from .clean import strip_tail, normalized
 
 
-def typesources():
-   "business types / abbreviations sorted by length of business type"
-   types = []
-   for business_type in terms_by_type:
-       for item in terms_by_type[business_type]:
-           types.append((business_type, item))
+def classifierList():
+    property_names = [p for p in dir(ElfEntry) if isinstance(
+        getattr(ElfEntry, p), property)]
+    return(property_names)
 
-   return sorted(types, key=lambda part: len(part[1]), reverse=True)
 
+def idClassification(abbreviation, source):
+    classifier_list = []
+    ElfCodeList = []
+    abbreviation = abbreviation
 
-def countrysources():
-   "business countries / type abbreviations sorted by length of type abbreviations"
-   countries = []
-   for country in terms_by_country:
-       for item in terms_by_country[country]:
-           countries.append((country, item))
+    for elf_code, values in Elf.items():
+        entity_codes = Elf[elf_code][0].local_abbreviations
+        if ";" in entity_codes:
+            entity_codes = entity_codes.split(';')
+        if abbreviation in entity_codes:
+            ElfCodeList.append(elf_code)
+        else:
+            pass
 
-   return sorted(countries, key=lambda part: len(part[1]), reverse=True)
+    for item in ElfCodeList:
+        test = getattr(Elf[item][0], source)
+        classifier_list.append(test)
+    # Unique values returned
+    myset = set(classifier_list)
+    classifier_list = list(myset)
```

Comment:
Do we really need to turn the set back to a list for some reason here? Set can be iterated too... Just curious.

Reply:
Technically no. It is returning a set for every possible match I believe. If there are multiple entries, it will get formatted in this way. They are referencing one "ELF" code, which could be considered a primary key of sorts. The issue that I was having is that a blank set would get returned first, so I'd need to figure out how to just get rid of that one.

Reply:
Using the example of:

matches("MyCompany Ltd", "country")

I get:

['Canada', 'United Kingdom', 'United Kingdom', 'United States of America'].

I need to be able to filter it down. If there's another way to do it, let me know.

Reply:
I am using set() now.